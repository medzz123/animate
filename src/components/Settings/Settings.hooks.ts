import { useCallback, useState } from 'react';

export const useSettingsState = () => {
  const [state, setState] = useState(false);

  const handlers = useCallback(
    () => ({ toggle: () => setState((s) => !s) }),
    []
  );

  return { state, handlers };
};
