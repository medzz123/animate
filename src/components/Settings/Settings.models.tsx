import { ThemeController } from '@theme/themeController';

export interface SettingsProps {
  themeController: ThemeController;
}
