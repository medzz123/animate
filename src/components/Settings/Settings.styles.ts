import { buttonReset } from '@theme/resets';
import { tokens } from '@theme/tokens';
import styled, { css } from 'styled-components';

export const SettingsContainer = styled.div<{ active: boolean }>`
  position: fixed;
  left: -${tokens.sizes['2xs']};
  top: 25%;
  display: flex;
  transition: transform ${tokens.transitions.fast} ease;
  ${(p) => p.active && `transform: translateX(${tokens.sizes['2xs']})`}
`;

export const SettingsContent = styled.div`
  width: ${tokens.sizes['2xs']};
  height: ${tokens.sizes['2xs']};
  background-color: ${(p) => p.theme.background};
  align-items: top;
  padding: ${tokens.sizes[16]};
`;

export const SettingsButton = styled.button<{ active: boolean }>`
  ${buttonReset}

  width: ${tokens.sizes[40]};
  height: ${tokens.sizes[40]};
  background-color: ${(p) => p.theme.accent};
  border-radius: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  transform-origin: center left;

  outline: none;

  transform: scale(0.7);
  opacity: 0.5;

  ${(p) =>
    p.active &&
    css`
      transform: scale(1);
      opacity: 1;
    `}

  &:hover {
    transform: scale(1);
    opacity: 1;
  }
`;

export const ColorContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: ${tokens.sizes[16]};
`;

export const Color = styled.div<{ color: string; active: boolean }>`
  margin-right: ${tokens.sizes[10]};
  width: ${tokens.sizes[28]};
  height: ${tokens.sizes[28]};
  background-color: ${(p) => p.color};

  ${(p) => p.active && `border: ${tokens.sizes[3]} solid #455a64;`}
`;
