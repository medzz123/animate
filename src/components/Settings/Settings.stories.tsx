import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Settings from './Settings';

export default {
  title: 'Components/Settings',
  component: Settings,
} as Meta;

const noop = () => {
  // Nothing
};

const Template: Story = (storyArguments) => (
  <Settings
    darkMode={{
      value: false,
      enable: noop,
      disable: noop,
      toggle: noop,
    }}
    {...storyArguments}
  />
);

export const SettingsStory = Template.bind({});
