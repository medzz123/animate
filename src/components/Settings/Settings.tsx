import React, { FunctionComponent } from 'react';
import { IoIosSettings } from 'react-icons/io';
import { Link } from 'react-router-dom';

import { themes } from '../../theme/theme';
import { useSettingsState } from './Settings.hooks';
import { SettingsProps } from './Settings.models';
import {
  Color,
  ColorContainer,
  SettingsButton,
  SettingsContainer,
  SettingsContent,
} from './Settings.styles';

const Settings: FunctionComponent<SettingsProps> = (props) => {
  const { state, handlers } = useSettingsState();
  const { themeController } = props;

  return (
    <SettingsContainer data-testid="settings" active={state}>
      <SettingsContent>
        <h3>Color switcher</h3>
        <ColorContainer>
          {Object.keys(themes).map((theme) => (
            <Color
              key={`theme-${theme}`}
              color={themes[theme].accent}
              active={themeController.themeName === theme}
              onClick={() => themeController.setTheme(theme)}
            />
          ))}
        </ColorContainer>
        <Link to="/testing-grounds">Testing Grounds</Link>
      </SettingsContent>
      <SettingsButton onClick={handlers().toggle} active={state}>
        <IoIosSettings size="32px" />
      </SettingsButton>
    </SettingsContainer>
  );
};

export default Settings;
