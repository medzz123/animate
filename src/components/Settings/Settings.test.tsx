import React from 'react';

import { render } from '../../../__tests__/testUtils';
import Settings from './Settings';

test('Settings Test', () => {
  const { getByTestId, container } = render(
    <Settings
      themeController={{
        theme: {
          background: '#232946',
          headline: '#fffffe',
          paragraph: '#b8c1ec',
          button: '#eebbc3',
          buttonText: '#232946',
        },
        state: false,
        toggleTheme: jest.fn(),
        mounted: true,
      }}
    />
  );

  const settingsNode = getByTestId('settings');

  expect(settingsNode).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
