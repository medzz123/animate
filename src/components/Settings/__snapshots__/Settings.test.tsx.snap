// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Test 1`] = `
<div>
  <div
    class="sc-bdvvaa bsYwbJ"
    data-testid="settings"
  >
    <div
      class="sc-gsDJrp fNLQom"
    >
      <h3>
        Color switcher
      </h3>
      <div
        class="sc-hKwCoD DzoLE"
      >
        <div
          class="sc-eCImvq hcChmD"
          color="#232946"
        />
        <div
          class="sc-eCImvq LTRcy"
          color="#004643"
        />
        <div
          class="sc-eCImvq cqtylu"
          color="#0f0e17"
        />
        <div
          class="sc-eCImvq dBqnlQ"
          color="#fffffe"
        />
        <div
          class="sc-eCImvq evAybb"
          color="#ffffff"
        />
        <div
          class="sc-eCImvq dgLjXR"
          color="#d4d8f0"
        />
      </div>
    </div>
    <button
      class="sc-dkPtyc dzRDgj"
    >
      <svg
        fill="currentColor"
        height="32px"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
        />
      </svg>
    </button>
  </div>
</div>
`;
